hcl/eval/eval_test.go

```diff
@@ -44,6 +44,36 @@ func TestEvalTmAbspath(t *testing.T) {
 	tempDir := t.TempDir()
 
 	for _, tc := range []testcase{
+		{
+			name: "tm_ternary - cond is true, with primitive values",
+			expr: `tm_ternary(true, "hello", "world")`,
+			want: want{
+				value: cty.StringVal("hello"),
+			},
+		},
+		{
+			name: "tm_ternary - cond is false, with primitive values",
+			expr: `tm_ternary(false, "hello", "world")`,
+			want: want{
+				value: cty.StringVal("world"),
+			},
+		},
+		{
+			name: "tm_ternary - cond is false, with partial not evaluated",
+			expr: `tm_ternary(false, local.var, "world")`,
+			want: want{
+				value: cty.StringVal("world"),
+			},
+		},
+		/*{
+			name: "tm_ternary - cond is true, with partial returning",
```

Comment:
ah the joys creating and comparing expressions on HCL :laughing:

Reply:
yeah I figured out this is impossible (at least in this case)... will have to remove this test.

Reply:
yay :laughing: